clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -268,11 +268,7 @@ public void updateEndOffsets(Map<TopicPartition, Long> newOffsets) {
     @Override
     public List<PartitionInfo> partitionsFor(String topic) {
         ensureNotClosed();
-        List<PartitionInfo> parts = this.partitions.get(topic);
-        if (parts == null)
-            return Collections.emptyList();
-        else
-            return parts;
+        return this.partitions.get(topic);
```

Comment:
Since MockConsumer is used in other cases as well, I think it would be safer to not change it here and let the unit test in Streams which is using MockConsumer to just timeout and throw the `StreamsException` in line 198.